### DISCUSSION TITLE
Using Views, Triggers & Functions Instead of Complex Client Code

### DISCUSSION THREAD

**Discussion by jtkeyva** [supabase/supabase#8452] [Questions] [answered]
Since we can create hosted views, when does it make sense to not use a view in a client app?
Since we can create hosted triggers & functions, when does it make sense to have database functions in the client app?

If most of the heavy lifting can be done on the server, do you make your clients as dumb as possible simply referencing views and calling RPC's? Seems especially ideal for cascaded many to many inserts.

Is anybody using this setup in production? What are your thoughts?

**Comment by steve-chavez** [comment, marked as answer]
> Is anybody using this setup in production?

Yes, there are users that only use functions in their public schema https://github.com/supabase/supabase/discussions/1360

> Since we can create hosted views, when does it make sense to not use a view in a client app?
> Since we can create hosted triggers & functions, when does it make sense to have database functions in the client app?

Those are a good practice as documented on https://postgrest.org/en/stable/schema_structure.html#schema-isolation

**Reply by jtkeyva** [reply]
@steve-chavez thank you!
